### PR TITLE
Simplify issue template

### PR DESCRIPTION
**Pull request by MichaelEischer**
What does this PR change? What problem does it solve?
-----------------------------------------------------
The PR simplifies the issue template by merging the `Steps to reproduce the behavior` and `How did you run restic exactly?` sections. These are closely related such that the actual description either ends up in one of them (https://github.com/restic/restic/issues/4447) or is partially duplicated across both sections (https://github.com/restic/restic/issues/4477).

In addition, it removes the `Do you have an idea how to solve the issue?` section. A user who submits an issue, usually has no chance at all to make a useful suggestion. Consequently, there is usually no suggestion or I just end up ignoring it. As this increases friction for users that report an issue, let's just remove it.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
No.

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- ~~[ ] I have added tests for all code changes.~~
- ~~[ ] I have added documentation for relevant changes (in the manual).~~
- ~~[ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).~~
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/Bug.md [hunk 1]
```diff
@@ -65,22 +72,12 @@ In this section, please try to concentrate on observations, so only describe
...
 Do you have any idea what may have caused this?
 -----------------------------------------------
 
-
-
-Do you have an idea how to solve the issue?
--------------------------------------------
-
+<!--
+Did something noteworthy happen on your system?
```
Comment (rawtaz) [inline .github/ISSUE_TEMPLATE/Bug.md:79]: Maybe `Did something noteworthy happen on your system, Internet connection, backend services, etc?`?

File: .github/ISSUE_TEMPLATE/Bug.md [hunk 2]
```diff
@@ -32,22 +32,29 @@ Output of `restic version`
 --------------------------
 
 
-How did you run restic exactly?
--------------------------------
+
+What backend/server/service did you use to store the repository?
```
Comment (rawtaz) [inline .github/ISSUE_TEMPLATE/Bug.md:36]: I'd probably skip the `server/` part here, but that's just me.

**Review by rawtaz — COMMENTED** [review]
Looks good overall, I added a couple of minor suggestions, other than that perhaps check that number of whitespace lines between each question is consistent.

File: .github/ISSUE_TEMPLATE/Bug.md (hunk 1 above)
Comment (MichaelEischer) [reply .github/ISSUE_TEMPLATE/Bug.md]: I've change the question accordingly.

File: .github/ISSUE_TEMPLATE/Bug.md (hunk 2 above)
Comment (MichaelEischer) [reply .github/ISSUE_TEMPLATE/Bug.md]: The server was part of the old section title. But I agree, that it's somewhat redundant.

**Review by rawtaz — APPROVED** [review]

**Pull request merged**